ivy.el

```diff
@@ -1946,9 +1950,9 @@ This is useful for recursive `ivy-read'."
               (t
                (push def coll))))
       (when sort
-        (if (and (functionp collection)
-                 (setq sort-fn (ivy--sort-function collection)))
-            (when (not (eq collection 'read-file-name-internal))
+        (if (functionp collection)
+            (when (and (not (eq collection 'read-file-name-internal))
+                       (setq sort-fn (ivy--sort-function collection caller)))
```

Comment:
Note that this changes the fall-through logic. Previously, the alternative (`else` branch) of the `if` conditional would have been taken if, for example, `(functionp collection)` was satisfied, but `(ivy--sort-function collection)` was not. Now, this combination causes neither of the conditional `if`'s branches to be taken. Is this intentional?